profiling.yaml

```diff
@@ -0,0 +1,112 @@
+scripts:
+  run-pidstat:
+  - js: ${{PIDSTAT_ENABLED}}
+    then:
+    - wait-for: HF_BENCHMARK_STARTED
+    - js: ${{HEROES_ENABLED}}
+      then:
+      - log: starting pidstat against ${{HOST.HEROES_REST_PID}}..
+      - sh: pidstat -p ${{HOST.HEROES_REST_PID}} ${{PIDSTAT_PARAMS}} > /tmp/HEROES_pidstat.log & export HEROES_PIDSTAT_PID="$!"
+      - queue-download: /tmp/HEROES_pidstat.log
+    - js: ${{VILLAINS_ENABLED}}
+      then:
+      - log: starting pidstat against ${{HOST.VILLAINS_REST_PID}}..
+      - sh: pidstat -p ${{HOST.VILLAINS_REST_PID}} ${{PIDSTAT_PARAMS}} > /tmp/VILLAINS_pidstat.log & export VILLAINS_PIDSTAT_PID="$!"
+      - queue-download: /tmp/VILLAINS_pidstat.log
+    - js: ${{LOCATIONS_ENABLED}}
+      then:
+      - log: starting pidstat against ${{HOST.LOCATIONS_GRPC_PID}}..
+      - sh: pidstat -p ${{HOST.LOCATIONS_GRPC_PID}} ${{PIDSTAT_PARAMS}} > /tmp/LOCATIONS_pidstat.log & export LOCATIONS_PIDSTAT_PID="$!"
+      - queue-download: /tmp/LOCATIONS_pidstat.log
+    - js: ${{FIGHTS_ENABLED}}
+      then:
+      - log: starting pidstat against ${{HOST.FIGHTS_REST_PID}}..
+      - sh: pidstat -p ${{HOST.FIGHTS_REST_PID}} ${{PIDSTAT_PARAMS}} > /tmp/FIGHTS_pidstat.log & export FIGHTS_PIDSTAT_PID="$!"
+      - queue-download: /tmp/FIGHTS_pidstat.log
+    - wait-for: HF_BENCHMARK_TERMINATED
+      then:
+      - sh: kill ${HEROES_PIDSTAT_PID} || echo "pidstat already stopped"
+      - sh: kill ${VILLAINS_PIDSTAT_PID} || echo "pidstat already stopped"
+      - sh: kill ${LOCATIONS_PIDSTAT_PID} || echo "pidstat already stopped"
+      - sh: kill ${FIGHTS_PIDSTAT_PID} || echo "pidstat already stopped"
+  
+  # measure RSS
+  run-pmap:
+  - js: ${{PMAP_ENABLED}}
+    then:
+    - wait-for: HF_BENCHMARK_STARTED
+    - js: ${{FIGHTS_ENABLED}}
+      then:
+      - queue-download: /tmp/FIGHTS_rss.log
+      - repeat-until: HF_BENCHMARK_TERMINATED
+        then:
+        - sleep: 1s
+        - sh: echo -n "$(date +"%I:%M:%S %p")  " >> /tmp/FIGHTS_rss.log
+        - sh:
+            # NOTE: with docker this requires starting the containers with the same host user, this is now sets by default with
+            # --user "$(id -u):$(id -g)"
+            command: pmap -x ${{HOST.FIGHTS_REST_PID}} ${{PMAP_PARAMS:}} 2>/dev/null | grep total >> /tmp/FIGHTS_rss.log
```

Comment:
This will create 2 lines per entry, one for timestamp, then another for the measurement. can we store in local variables and write one line to file.

Reply:
Technically they should be on one line already as the timestamp is echoed with `-n` option which should not create a new line at the end. But I think we can still save it in a variable